File: nodejs/fiWareConnector.js
"use strict";

//################# LIBRARIES ################   
var request = require("request");

//################ FIWARE VARIABLES ################

var FIWARE_APIKEY = "apikey" ;


//Device´s name
var FIWARE_DEVICE = "deviceName" ;
var FIWARE_SERVER = 'fiwareServer';
var FIWARE_PORT = 'port';


var STATUS_CODE_OK = 200;

var postMeasures = function(measures){

var body=""; //FIWARE IoT Stack body message
var counter = Object.keys(measures).length; //number of sensors
	
for (var sensor in measures) { 
	body = body + sensor + '|' + measures[sensor];
	//add measures separator "#" to the body only when it is not the last measure
	if(counter>1){
		body = body +"#";
		counter--;
	}
}
    
console.log('--Sending measures to FIWARE IoT Stack '+body);



var options = { 
  method: 'POST',
  url: 'http://'+FIWARE_SERVER+':'+FIWARE_PORT+'/iot/d',
  qs: { i: FIWARE_DEVICE, k: FIWARE_APIKEY },
  body: body 
};

request(options, function (error, response, body) {
	if (error){
		console.log('--ERROR connectivity problems');	
	}else{
		if (response.statusCode == STATUS_CODE_OK){
			console.log('--Data uploaded to FIWARE IoT Stack');
		}else{
			console.log('--ERROR Status Code: '+response.statusCode);
		}
	}
});
	
};


module.exports = {
    postMeasures: postMeasures
};

File: python/fiwareConnector.py
#!/usr/bin/python
#This is an adaptation of the code: https://github.com/telefonicaid/fiware-edison/tree/develop/python
import requests
import json

#Time between measures
MEASURES_PERIOD = 2
FIWARE_SERVER = "YOUR_FIWARE_SERVER_ADDRESS"
FIWARE_PORT = "xxxx"
FIWARE_APIKEY = "xxxxxxx"
FIWARE_DEVICE = "myEdison"
class FiwareConnector:
################ FIWARE VARIABLES ################
    @staticmethod
    def postMeasures(measures):

        first = True
        payload = ""
        received_commands=[]
        #Parse chain of commands
        for key,value in measures.items():
            #Measures payload format is alias1|value1#alias2|value2#....
            #Add measures separator "#" to the request payload only when it is not the first measure
            if(first):
                first = False
            else:
                payload = payload + "#"

            payload = payload + key + "|" + value

        print('Sending measures to FIWARE IoT Stack '+payload)

        url = "http://" + FIWARE_SERVER+ ":" +FIWARE_PORT+ "/iot/d"
        querystring = {"i": FIWARE_DEVICE, "k": FIWARE_APIKEY, "getCmd":"1"}
        try:
            r = requests.post(url, data=payload, params=querystring)
            print "Response Status Code: "+str(r.status_code)
        except:
            print "Error chek your URL or Connection, please"
